Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/Brakes/MSTS/AirSinglePipe.cs

```diff
@@ -448,6 +449,14 @@ public override void Update(float elapsedClockSeconds)
                     AuxResPressurePSI += dp;
                     BrakeLine1PressurePSI -= dp * AuxBrakeLineVolumeRatio;  // Adjust the train brake pipe pressure
                 }
+                if (AuxResPressurePSI > BrakeLine1PressurePSI) // Allow small flow from auxiliary reservoir to brake pipe to restore normal pressure after overcharge
```

Comment:
Given the fact that overcharging is not present in all air brake systems, I would suggest that an additional "switching factor" be used.

For example, something like "if (AuxResPressurePSI > BrakeLine1PressurePSI && OverChargingActive)"

where OverChargingActive = a parameter that is set when the user enters overcharging parameters in the ENG/WAG file.

Reply:
Is there any technical reason which justifies not having that reverse flow in all triple valves? It is so small that shouldn't affect brake application, and only occurs if valve is in release position.

Reply:
Without studying a wide range of triple valve types, I can't confirm that this leakage occurs. Certainly the documentation that I have looked at quickly does not appear to support overcharging, and hence talks about the Aux Res flow stopping when it is equal to the train pipe pressure.

The size of flow is immaterial, as philosophically I would prefer to apply functionality only to the features that are applicable to a particular equipment type. So as not all triple valves appear to have supported overcharging, I would still suggest that the feature only be enabled when overcharging is required rather then generically across all types of equipment.

As an aside, given the variation in operation of the different brake components used over the years, I would prefer to see documentation on how to set up a particular brake type, as this would assist OR users and eliminate confusion. For example, an A6-ET brake is different to a 26-L, etc. Hence some of the tokens, and "other variations" will need to be defined and coded differently. So it would be good if your brake system was described, and then coded as required.

Reply:
This leakage is more related to brake insensitivity than to overcharge. I haven't been able to found any document for triple valves, but for distributors UIC regulation says that brakes shouldn't be applied if brake pipe pressure is decreased at a lower rate than 0.3 bar in 60 seconds. What do you think about creating a `Wagon(ORTSBrakeInsensitivity(` parameter to regulate this flow? Note that this brake insensitivity is also needed in order to not apply brakes due to a small brake pipe leakage.

Reply:
> "I haven't been able to found any document for triple valves"

I looked at a number of manuals when trying to add some features to the brake code. Here is but one example -

https://archive.org/details/westinghouseairb00inteuoft

This publication describes a number of different types of triple valves. Naturally there maybe more modern or non US types of valves which will have variations to those described.

By the time the different triple valves, brake control valves, other functions, such as load drop compensation, etc are considered, the number of variations can be reasonably large.

> "Note that this brake insensitivity is also needed in order to not apply brakes due to a small brake pipe leakage."

This is perhaps a more modern feature that may not have been present in older systems? In non self-lapping systems some level of leakage occurred in the brake pipe, and this gradually allowed the brakes to be applied.

Overall, my personal feeling is that the current "default" brake functionality should remain untouched as it provides a "reasonable" base foundation. Extra functionality can then be added on top of it to customise the operation to a particular equipment type or era.

My concern was not with the functionality that you were proposing (as I assumed that you had done sufficient research to ensure that it was correct), but more with the fact that it was also possibly changing the base functionality.

> "What do you think about creating a Wagon(ORTSBrakeInsensitivity( parameter to regulate this flow?"

I have no problems with this if you feel it is necessary for the functionality that you are adding, and it is an addon on top of the base functionality, rather then a change to the base code. 

If you wish to discuss anything other information in detail, how about sending me a PM?

Reply:
Thanks for the link! It's really useful.
I have added the parameter in order to keep things as they were by default.
If I have more doubts I'll write you.